src/Main.cpp

```diff
@@ -1,49 +1,77 @@
+#include <genset_whisperpower_ddc/VariableSpeed.hpp>
+#include <genset_whisperpower_ddc/VariableSpeedMaster.hpp>
 #include <iostream>
+#include <jsoncpp/json/json.h>
 #include <list>
 #include <memory>
 #include <signal.h>
-#include <genset_whisperpower_ddc/VariableSpeed.hpp>
-#include <genset_whisperpower_ddc/VariableSpeedMaster.hpp>
-#include <genset_whisperpower_ddc/ControlCommand.hpp>
 
 using namespace std;
 using namespace genset_whisperpower_ddc;
 
 bool stop = false;
 
-void usage(ostream& stream) {
+void usage(ostream& stream)
+{
     stream << "usage: genset_whisperpower_ddc_ctl URI CMD\n"
            << "where:\n"
            << "    URI is a iodrivers_base URI\n"
            << "\n"
            << "Available Commands\n"
-           << "    run: send start control command to the DDC GenVerter power controller and then keep it running by "
-           << "repeatedly sending the keep_alive command until the user press any key. When this happens, send the stop command to stop de generator.\n"
+           << "    run: send start control command to the DDC GenVerter power controller "
+              "and then keep it running by "
+           << "repeatedly sending the keep_alive command until the user press any key. "
+              "When this happens, send the stop command to stop de generator.\n"
            << "    read_frame: read a frame and print its content\n"
+           << "    speed-info: export parameters in a json file.\n"
            << endl;
 }
 
-void handler(int s) {
+void handler(int s)
+{
     stop = true;
 }
 
-Frame waitForValidFrame(std::unique_ptr<VariableSpeedMaster> &master) {
+Frame waitForValidFrame(unique_ptr<VariableSpeedMaster>& master)
+{
     while (true) {
         try {
             Frame frame = master->readFrame();
-            if (frame.targetID == variable_speed::PANELS_ADDRESS && frame.sourceID == variable_speed::DDC_CONTROLLER_ADDRESS) {
+            if (frame.targetID == variable_speed::PANELS_ADDRESS &&
+                frame.sourceID == variable_speed::DDC_CONTROLLER_ADDRESS) {
                 return frame;
             }
         }
-        catch(const variable_speed::WrongSize& e) {}
-        catch(const variable_speed::InvalidChecksum& e) {}
-        // iodrivers_base may throw this error when receiving a SIGINT, but it can be ignored
-        catch(const iodrivers_base::UnixError& e) {}
+        catch (const variable_speed::WrongSize& e) {
+        }
+        catch (const variable_speed::InvalidChecksum& e) {
+        }
+        // iodrivers_base may throw this error when receiving a SIGINT, but it can be
+        // ignored
+        catch (const iodrivers_base::UnixError& e) {
+        }
     }
     // Never reached
 }
 
-int main(int argc, char** argv) {
+void writeAndExportParameters(GeneratorState genset_state)
+{
+    Json::Value params;
+
+    params["state"]["rotation speed"] = genset_state.rotation_speed;
+    params["state"]["time"] = (int)genset_state.time.toMilliseconds();
```

Comment:
prefer cpp style casting 

```suggestion
    params["state"]["time"] = static_cast<int>(genset_state.time.toMilliseconds());
```